src/components/cdTab.js

```diff
@@ -1,35 +1,187 @@
+import {
+  Alert,
+  Box,
+  FormControlLabel,
+  FormGroup,
+  Switch,
+  Tooltip,
+} from '@mui/material'
 
 import MultilineTextViewer from './util/MultilineTextViewer'
+import { useFetchComponentDescriptor } from '../api/useFetch'
+import CenteredSpinner from './util/CenteredSpinner'
+import { SearchParamContext } from '../App'
 import { toYamlString } from '../util'
 
 
+const FetchAndView = ({
+  componentName,
+  componentVersion,
+  ocmRepo,
+  versionFilter,
+  setRaw,
+}) => {
+  // eslint-disable-next-line no-unused-vars
+  const [componentDescriptor, isLoading, isError, _] = useFetchComponentDescriptor({
+    componentName: componentName,
+    ocmRepoUrl: ocmRepo,
+    version: componentVersion,
+    versionFilter: versionFilter,
+    raw: true,
+  })
+
+  React.useEffect(() => {
+    setRaw(componentDescriptor)
+  }, [setRaw, componentDescriptor])
+
+  return <Viewer
+    isError={isError}
+    componentDescriptor={JSON.stringify(componentDescriptor, null, 2)}
+  />
+}
+FetchAndView.displayName = 'FetchAndView'
+FetchAndView.propTypes = {
+  componentName: PropTypes.string.isRequired,
+  componentVersion: PropTypes.string.isRequired,
+  ocmRepo: PropTypes.string,
+  versionFilter: PropTypes.string,
+  setRaw: PropTypes.func.isRequired,
+}
+
+
+const Options = ({
+  isLoading,
+  showRaw,
+  setShowRaw,
+}) => {
+  return <Box>
+    <FormGroup>
+      <FormControlLabel
+        control={
+          <Tooltip
+            title='Raw document as retrieved from OCM repository, without (de)serialisation'
+          >
+            <Switch
+              checked={showRaw}
+              // eslint-disable-next-line no-unused-vars
+              onChange={(event) => setShowRaw((prev) => !prev)}
+              disabled={isLoading}
+            />
+          </Tooltip>
+        }
+        label='show raw'
+      />
+    </FormGroup>
+  </Box>
+}
+Options.displayName = 'Options'
+Options.propTypes = {
+  isLoading: PropTypes.bool,
+  showRaw: PropTypes.bool.isRequired,
+  setShowRaw: PropTypes.func.isRequired,
+}
+
+
+const Viewer = ({
+  componentDescriptor,
+  isError
+}) => {
+  if (isError) return <Alert severity='critical'>
+    Something went wrong
+  </Alert>
+
+  return <MultilineTextViewer
+    text={componentDescriptor}
+  />
+}
+Viewer.displayName = 'Viewer'
+Viewer.propTypes = {
+  componentDescriptor: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+  isError: PropTypes.bool,
+}
+
+
+const ComponentDescriptor = ({
+  showRaw,
+  raw,
+  componentDescriptor,
+  setRaw,
+  ocmRepo,
+  versionFilter,
+}) => {
+  return (showRaw & !raw) // prevent re-fetching if raw cd is present already
+    ? <FetchAndView
+      componentName={componentDescriptor.component.name}
+      componentVersion={componentDescriptor.component.version}
+      ocmRepo={ocmRepo}
+      versionFilter={versionFilter}
+      setRaw={setRaw}
+    />
+    : <Viewer
+      isError={false} // handled earlier already, ok to hard code here
```

Comment:
As the `Viewer` component only contains error handling + the `MultilineTextViewer`, _and_ the error is hardcoded in 1 out of 2 occurrences, why not removing the `Viewer` wrapper and directly use `MultilineTextViewer` here and move error handling into `FetchAndView`?